Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -119,6 +119,10 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
 
   var commandsByFile: [DocumentURI: SwiftCompileCommand] = [:]
 
+  var documentParseTransition: [DocumentURI: (previousTree: SourceFileSyntax, collector: IncrementalParseNodeAffectRangeCollector)] = [:]
```

Comment:
We already store the tree in `DocumentSnapshot`. I think the cleaner design would be to also store the `IncrementalParseNodeAffectRangeCollector` there instead of adding this dictionary to `SwiftLanguageServer`.